marshal.go

```diff
@@ -144,3 +145,157 @@ func (i InstanceMetadata) MarshalXML(e *xml.Encoder, start xml.StartElement) err
 	// flush to ensure tokens are written
 	return e.Flush()
 }
+
+// startLocalName creates a start-tag of an XML element with the given local name and no namespace name.
+func startLocalName(local string) xml.StartElement {
+	return xml.StartElement{Name: xml.Name{Space: "", Local: local}}
+}
+
+type metadataMap map[string]string
+
+// MarshalXML is a custom XML marshaler for metadataMap, mapping each metadata name/value pair to a
+// correspondingly named XML element with the pair's value as character data content.
+func (m metadataMap) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	if err := e.EncodeToken(start); err != nil {
+		return err
+	}
+
+	for k, v := range m {
+		if err := e.EncodeElement(v, startLocalName(k)); err != nil {
+			return err
+		}
+	}
+
+	return e.EncodeToken(start.End())
```

Comment:
Should we call Flush() here?
https://golang.org/pkg/encoding/xml/#Encoder.EncodeToken